File: calc.py
import sys
from antlr4 import *
from labeldExprLexer import labeldExprLexer
from labeldExprParser import labeldExprParser
from EvalVisitor import EvalVisitor

def main(argv):
    if len(argv) > 1:
        input_stream = FileStream(argv[1])
    else:
        input_stream = InputStream(sys.stdin.read())

    lexer = labeldExprLexer(input_stream)
    stream = CommonTokenStream(lexer)
    parser = labeldExprParser(stream)
    tree = parser.prog()

    visitor = EvalVisitor()
    visitor.visit(tree)

if __name__ == '__main__':
    main(sys.argv)

File: Calc.java
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.*;


public class Calc {
public static void main(String[] args) throws Exception {
java.io.InputStream is = System.in;
ANTLRInputStream input = new ANTLRInputStream(is);
labeldExprLexer lexer = new labeldExprLexer(input);
CommonTokenStream tokens = new CommonTokenStream(lexer);
labeldExprParser parser = new labeldExprParser(tokens);
ParseTree tree = parser.prog();
EvalVisitor eval = new EvalVisitor();
eval.visit(tree);
}
}
